interfaces/builtin/iio_test.go

```diff
@@ -0,0 +1,184 @@
+
+	// Mock for Gadget Snap
+	gadgetSnapInfo := snaptest.MockInfo(c, `
+name: some-device
+type: gadget
+slots:
+  test-udev-1:
+    interface: iio
+    path: /dev/iio:device1
+  test-udev-2:
+    interface: iio
+    path: /dev/iio:device2
+  test-udev-3:
+    interface: iio
+    path: /dev/iio:device0
+  test-udev-bad-value-1:
+    interface: iio
+    path: /dev/iio
+  test-udev-bad-value-2:
+    interface: iio
+    path: /dev/iio:devicea
+  test-udev-bad-value-3:
+    interface: iio
+    path: /dev/iio:device2a
+  test-udev-bad-value-4:
+    interface: iio
+    path: /dev/foo-0
+  test-udev-bad-value-5:
+    interface: iio
+    path: /dev/iio:devicefoo
+  test-udev-bad-value-6:
+    interface: iio
+    path: ""
+  test-udev-bad-value-7:
+    interface: iio
+  test-udev-bad-interface-1:
+    interface: other-interface
+`, nil)
+	s.testUdev1 = &interfaces.Slot{SlotInfo: gadgetSnapInfo.Slots["test-udev-1"]}
+	s.testUdev2 = &interfaces.Slot{SlotInfo: gadgetSnapInfo.Slots["test-udev-2"]}
+	s.testUdev3 = &interfaces.Slot{SlotInfo: gadgetSnapInfo.Slots["test-udev-3"]}
+	s.testUdevBadValue1 = &interfaces.Slot{SlotInfo: gadgetSnapInfo.Slots["test-udev-bad-value-1"]}
+	s.testUdevBadValue2 = &interfaces.Slot{SlotInfo: gadgetSnapInfo.Slots["test-udev-bad-value-2"]}
+	s.testUdevBadValue3 = &interfaces.Slot{SlotInfo: gadgetSnapInfo.Slots["test-udev-bad-value-3"]}
+	s.testUdevBadValue4 = &interfaces.Slot{SlotInfo: gadgetSnapInfo.Slots["test-udev-bad-value-4"]}
+	s.testUdevBadValue5 = &interfaces.Slot{SlotInfo: gadgetSnapInfo.Slots["test-udev-bad-value-5"]}
+	s.testUdevBadValue6 = &interfaces.Slot{SlotInfo: gadgetSnapInfo.Slots["test-udev-bad-value-6"]}
+	s.testUdevBadValue7 = &interfaces.Slot{SlotInfo: gadgetSnapInfo.Slots["test-udev-bad-value-7"]}
+	s.testUdevBadInterface1 = &interfaces.Slot{SlotInfo: gadgetSnapInfo.Slots["test-udev-bad-interface-1"]}
+
+	// Snap Consumers
+	consumingSnapInfo := snaptest.MockInfo(c, `
+name: client-snap
+plugs:
+  plug-for-port-1:
+    interface: iio
+apps:
+  app-accessing-1-port:
+    command: foo
+    plugs: [iio]
+`, nil)
+	s.testPlugPort1 = &interfaces.Plug{PlugInfo: consumingSnapInfo.Plugs["plug-for-port-1"]}
+}
+
+func (s *IioInterfaceSuite) TestName(c *C) {
+	c.Assert(s.iface.Name(), Equals, "iio")
+}
+
+func (s *IioInterfaceSuite) TestSanitizeBadGadgetSnapSlot(c *C) {
+
+	err := s.iface.SanitizeSlot(s.testUdevBadValue1)
+	c.Assert(err, ErrorMatches, "iio path attribute must be a valid device node")
+
+	err = s.iface.SanitizeSlot(s.testUdevBadValue2)
+	c.Assert(err, ErrorMatches, "iio path attribute must be a valid device node")
+
+	err = s.iface.SanitizeSlot(s.testUdevBadValue3)
+	c.Assert(err, ErrorMatches, "iio path attribute must be a valid device node")
+
+	err = s.iface.SanitizeSlot(s.testUdevBadValue4)
+	c.Assert(err, ErrorMatches, "iio path attribute must be a valid device node")
+
+	err = s.iface.SanitizeSlot(s.testUdevBadValue5)
+	c.Assert(err, ErrorMatches, "iio path attribute must be a valid device node")
```

Comment:
It's unclear why we have all of these tests. They are all broken in the same way, and we have no tests at all that verify the proper matching string.